src/datumaro/plugins/data_formats/kaggle/base.py

```diff
@@ -42,71 +47,135 @@ def __init__(
         super().__init__(ctx=ctx)
 
         self._subset = subset
-
         self._path = path
-        self._columns = columns
 
-        self._items, label_cat = self._load_items(ann_file, columns)
-        self._categories = {AnnotationType.label: label_cat}
+        if "media" not in columns:
+            raise MissingFieldError("media")
+
+        self._label_cat = LabelCategories()
+        self._items = self._load_items(ann_file, columns)
+        self._categories = {AnnotationType.label: self._label_cat}
+
+    def _get_media_path(self, media_name: str):
+        if not media_name.lower().endswith(tuple(IMAGE_EXTENSIONS)):
+            for ext in IMAGE_EXTENSIONS:
+                media_path = osp.join(self._path, media_name + ext)
+                if osp.exists(media_path):
+                    return media_path
+            return None
+        else:
+            return osp.join(self._path, media_name)
+
+    def _parse_bbox_coords(self, bbox_str):
+        bbox_str = bbox_str.strip()
+
+        # Check if the string starts with "[" and ends with "]"
+        if bbox_str.startswith("[") and bbox_str.endswith("]"):
+            bbox_str = bbox_str[1:-1]
+
+        coords = list(filter(None, re.split(r"\s|,", bbox_str)))
+
+        # expected to output [x1, y1, x2, y2]
+        return [float(coord.strip()) for coord in coords]
+
+    def _load_annotations(self, datas: list, indices: Dict[str, int], bbox_flag: bool):
+        if "label" in indices:
+            label_name = str(datas[indices["label"]])
+            label, cat = self._label_cat.find(label_name)
+            if not cat:
+                self._label_cat.add(label_name)
+                label, _ = self._label_cat.find(label_name)
+        else:
+            _, cat = self._label_cat.find("object")
+            if not cat:
+                self._label_cat.add("object")
+            label = 0
+
+        if "label" in indices and not bbox_flag:
+            return Label(label=label)
+        elif bbox_flag:
```

Comment:
```suggestion
        if bbox_flag:
```
Through check label_index above, we can reduce this if clause.

Reply:
fixed at fc8b3ed6b7f04414afa45cf0219e81343bd9e61d